### PR TITLE
fix(fast_check): properly handle rest parameters in overloads

### PR DESCRIPTION
**Pull request by lucacasonato**
Fixes https://github.com/jsr-io/jsr/issues/376

### CODE REVIEW

File: tests/specs/graph/jsr/FastCheck_Overloads.txt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  argsToArray(..._args: string[]): void;
+  argsToArray(..._args: number[]): void;
+  argsToArray(..._args: unknown[]): void {}
+}
+
+function overloadSpread(...args: string[]);
+function overloadSpread(...args: number[]);
+export function overloadSpread(...args: unknown[]) {
+  return args;
+}
```
Comment (marvinhagemeister) [inline tests/specs/graph/jsr/FastCheck_Overloads.txt:18]: Mixing non-exported overloads with an exported implementation is not valid in TS. See https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwDYEMDOa4qgczQFQgEEooUBPOAbwCg4tcDjSKAKAOk4H1s8AuOGhhQAlgDscAbQC6ASgEA3CCIAmAbjoM8hEmXIduvNALEBXALYAjYFBny4S1RvpGdLfZ3Y9GA02IDWYhAA7mJ2isoq1AC+NLE0AGZ+CDAiEGJwEAo2SBAoKgDKYFDA+QbsRgJCohLhWGLkGkliKWkZWTl5hcWlKuWVcGZWNnUoDRqgkLBwza3pmdlQuflFJWWeA36BIWFyAmOUtPQlMKZQGUYa0UA

File: tests/specs/graph/jsr/FastCheck_Overloads.txt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    return {} as never;
+  }
+  --- DTS ---
+  export declare class argsToArray {
+    argsToArray(..._args: string[]): void;
+    argsToArray(..._args: number[]): void;
+  }
+  declare function overloadSpread(...args: string[]);
+  declare function overloadSpread(...args: number[]);
+  export declare function overloadSpread(...param0?: any): any;
```
Comment (marvinhagemeister) [inline tests/specs/graph/jsr/FastCheck_Overloads.txt:89]: The implementation is not included in the dts output from `tsc` as it cannot be referenced. Might have to do with the overloads not being exported.

File: tests/specs/graph/jsr/FastCheck_Overloads.txt (hunk 1 above)
Comment (lucacasonato) [reply tests/specs/graph/jsr/FastCheck_Overloads.txt]: whoops - you are right. fixed

File: tests/specs/graph/jsr/FastCheck_Overloads.txt (hunk 2 above)
Comment (lucacasonato) [reply tests/specs/graph/jsr/FastCheck_Overloads.txt]: yup that was it, fixed.

**Review by marvinhagemeister — APPROVED** [review]

**Comment by atjn** [comment]
Wow that was a fast turnaround. Thank you very much!

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**